internal/declarative/v2/reconciler.go

```diff
@@ -360,6 +360,10 @@ func (r *Reconciler) syncResources(ctx context.Context, clnt Client, manifest *v
 		}
 	}
 
+	if !manifest.GetDeletionTimestamp().IsZero() {
+		return r.setManifestState(manifest, shared.StateDeleting)
```

Comment:
but inside this `setManifestState`, we did same thing, set state to deleting when manifest has deletiontimestamp

https://github.com/kyma-project/lifecycle-manager/blob/main/internal/declarative/v2/reconciler.go#L418-L420

Reply:
but consider rename this `setManifestState`, it's really misleading, if it's a `set` method, I don't expect the input `state` will be modified in the body of the function.

Reply:
Good catch, thanks! I removed modifying the state arg in `setManifestState`. We need to check the deletion timestamp outside because we can't reliably determine the manager state if there is a deletion timestamp. And this way it is now also consistent. We first determine the state to transition to, then we call `setManfiestState` with that state.

TBH, I am also a bit confused about the rest of the `setManifestState` method. If I understand it right, we set Condition `Installation` to `True` and set Last Operator Message to `installation is ready and resources can be used` whenever we transition between states other than Processing. So if we go from Reay to Warning, we will keep this condition and message. If we go from Ready to Error, we will keep this condition and message, ... Is this expected or do we need to straighten out a few things?

Reply:
Yes, the rest code is a bit bluring, I think we can improve it, but logically should fine I think.

When the reconcilation enter into `setManifestState`, we don't have `Error` state case, this should already requeued in previous steps.

So the rest state are `Ready`, `Processing`, `Warning`, basically all of these state means installation should be finished.

`if !meta.IsStatusConditionTrue(status.Conditions, installationCondition.Type) || status.State != state `
this meaning:
1. the actual manifest CR status in cluster (`status`), the  `Installation` condition is false; basically this means the actual installation condition in cluster is false.
2. or actual manifest CR state (` status.State`) is different than the state to be updated (`state`); this means we need to update the state,

in both case, we should update the installation condition to true.

Reply:
Thanks. I have created a follow-up to investigate this more thoroughly: https://github.com/kyma-project/lifecycle-manager/issues/1846

Right now I don't know if I understand the setting of those conditions correctly. Let's discuss that.

But until then, we could merge this one as proposed if okay from your side.

Reply:
this logic prevents all follow steps, right? if Manifest CR is under deletion, it always requeue with set state to `Deleting`

I think you need to update `if !manifest.GetDeletionTimestamp().IsZero() && manifest.status.state != "deleting"`